### DISCUSSION TITLE
Request for new feature

### DISCUSSION THREAD

**Discussion by Sumanadhikari72** [mush42/bookworm#100] [Ideas]
I want to request to impliment translate feature. Sometime, we need this to understand text deeply. As every people can't be perfect in different language through out the world, translation can be a best way. Many people like me feel comfortable with local language. Small example, I can understand English but still not like nepali because I am from Nepal and my mother tongue is nepali.

**Comment by mush42** [comment]
Hi @Sumanadhikari72 

I hear you!

Are there any translation APIs that Bookworm can use?

I think that there is no any free translation service that offers high-quality translations.

But if you're interested in an English-to-English translation, we got you covered. I'm currently contemplating adding the ability to download and use Wiktionary.

Best

**Comment by Sumanadhikari72** [comment]
I have also limited idea regarding this. I found Apertium, , Tatoeba, LibreTranslate and so on but, I don't know how it' work. If you find useful than try to add. Are google translator, MS translator and Yandex translator totally paid?

**Comment by cary-rowen** [comment]
Hi @mush42 

As far as I know google translator has a free api but google detects frequent requests will limit access, the limit will be restored within a few hours.
The unrestricted google translate api needs to calculate a tk but I don't know if it's legal to discuss the topic here.

Cheers

**Comment by NiKola-UE** [comment]
Hello for all. 



I would also join this discussion if it is not latter for it. 



The ability to translate documents is a really great thing, but there is still no translator or dictionary that allows an unlimited number of characters. Some incomplete lists of the open source translators and dictionaries are 
[here](https://alternativeto.net/category/education-and-reference/language-translation/?license=opensource) 
and 
[here](https://alternativeto.net/category/education-and-reference/dictionary/?license=opensource), 
so savvy developers can do their best. Also, the site 
machinetranslation.com 
also can be useful and helpful. 




My proposal is the ability to convert and export documents from one supported format to others, which will reduce the need for additional demand for more complicated, insecure and commercial programs... 




Consider adding some of this to the new version of Bookworm that is about to finally come out.
